variants/examples/virtual_fct_port.cpp

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2025 CExA-project
+// SPDX-License-Identifier: MIT or Apache-2.0 with LLVM-exception
+//
+// Example of porting code using virtual functions to Kokkos by using variants
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Variant.hpp>
+
+#include <iostream>
+
+// Example of a code using virtual function, we want to parallelize its main
+// loop using Kokkos
+namespace original {
+struct A {
+  virtual int f(int in) { return 4 + in; }
+};
+struct B : A {
+  int f(int in) { return 6 + in; }
+};
+struct C : A {
+  int f(int in) { return 2 * A::f(in); }
+};
+
+void do_computation() {
+  std::vector<A*> v(3);
+  v[0] = new A;
+  v[1] = new B;
+  v[2] = new C;
+
+  for (int i = 0; i < 3; ++i) {
+    // This won't work on GPU (see here
+    // https://kokkos.org/kokkos-core-wiki/ProgrammingGuide/Kokkos-and-Virtual-Functions.html)
+    std::cout << i << ": " << v[i]->f(i) << "\n";
+  }
+  
+  for (int i = 0; i < 3; i++) {
+    delete v[i];
+  }
```

Comment:
Just to match the rest of the code
```suggestion
  for (int i = 0; i < 3; ++i) {
    delete v[i];
  }
```